code-collab-app/src/storage/peer_data.js

```diff
@@ -0,0 +1,142 @@
+
+            //  inform tracker new session info
+            this.socket.emit('new session', JSON.stringify(peer_info));
+
+            this.socket.on('session created', function(session_info){
+                let temp = JSON.parse(session_info);
+                this.current_session = temp.session_name;
+                this.current_session_id = temp.session_id;
+            });
+
+            this.listen_for_req();
+        })
+    },
+
+    //  Join an active session or open preexisted doc
+    join_session(session_name){
+        this.current_session = session_name;
+
+        this.peer = new Peer();
+
+        this.peer.on('open', function(id){
+            this.peer_id = id;
+            console.log("id: " + this.id);
+
+            let peer_info = {   "session_name": session_name, 
+                                "peer_id": this.peer_id};
+
+            //  Request list of peers in <session_name> session
+            this.socket.emit('join session', JSON.stringify(peer_info));
+
+            //  wait for a list of peers currently in the session
+            this.socket.on('peer_list', function(list){
+                this.session_peers = JSON.parse(list);
+                this.request_doc_data();
+            });
+
+            this.listen_for_req();
+        });
+    },
+
+    //  request for current session's doc data 
+    //  from a fellow peer
+    request_doc_data(){
+        //  check the number of peers in current session
+        //  if == 0, then no need to connect to other peers
+        if(this.session_peers.length !== 0){
+            //  connect with all peers
+            this.session_peers.forEach(function(p, i){
+                this.session_peers_conn[i] = this.peer.connect(p);
+            });
+
+            //  request doc data from one of them
+            this.session_peers_conn[0].on('open', function(){
+                this.session_peers_conn[0].send('get doc');
+            });
+
+            this.session_peers_conn[0].on('error', function(err){
+                console.log(err);
+            });
+
+            this.session_peers_conn[0].on('data', function(data){
+                console.log(data);
+            });
+        }
+    },
+
+    //  listen for other peer's initial connection
+    //  and their request for the doc data
+    listen_for_req(){
+        this.peer.on('connection', function(dataConnection) { 
+            dataConnection.on('error', function(err){
+                console.log("listen for req err: " + err);
+            });
+
+            dataConnection.on('data', function(data){
+                dataConnection.send(this.doc_data);
+            });
+        });
+    },
+
+    upload_document(d){
+        this.doc_data = d;
+    },  
+
+    get get_doc_data(){
+        return this.doc_data;
+    },
+
+    get get_session_list(){
+        return this.session_list;
+    },
+
+    get get_peer_id(){
+        return this.peer_id;
+    }
```

Comment:
The variables in a mobx store (`doc_data`, `session_list`, `peer_id`, etc) are publicly accessible so don't worry about adding get functions for every variable the UI needs. Mobx `get` functions are only there if we need to do some pre-processing before passing it to the UI.

Reply:
Sounds good, thank you!